SerialPrograms/Source/PokemonSV/Programs/PokemonSV_Navigation.cpp

```diff
@@ -398,6 +399,120 @@ void leave_box_system_to_overworld(const ProgramInfo& info, ConsoleHandle& conso
 }
 
 
+void open_pokedex_from_overworld(const ProgramInfo& info, ConsoleHandle& console, BotBaseContext& context){
+    {
+        OverworldWatcher overworld(COLOR_CYAN);
+        context.wait_for_all_requests();
+        int ret = wait_until(
+            console, context,
+            std::chrono::seconds(10),
+            {overworld}
+        );
+        context.wait_for(std::chrono::milliseconds(100));
+        if (ret == 0){
+            console.log("Detected overworld.");
+            pbf_press_button(context, BUTTON_MINUS, 20, 105); // open pokedex
+        }
+        else{
+            throw OperationFailedException(
+                ErrorReport::SEND_ERROR_REPORT, console,
+                "open_pokedex_from_overworld(): No overworld state found after 10 seconds.",
+                true
+            );
+        }
+    }
+
+    WallClock start = current_time();
+    while (true){
+        if (current_time() - start > std::chrono::minutes(1)){
+            throw OperationFailedException(
+                ErrorReport::SEND_ERROR_REPORT, console,
+                "open_pokedex_from_overworld(): Failed to open map after 1 minute.",
+                true
+            );
+        }
+
+        OverworldWatcher overworld(COLOR_CYAN);
+        MapExitWatcher pokedex(COLOR_RED);
+
+        context.wait_for_all_requests();
+        int ret = wait_until(
+            console, context,
+            std::chrono::seconds(30),
+            {overworld, pokedex}
+        );
+        context.wait_for(std::chrono::milliseconds(100));
+        switch (ret){
+        case 0:
+            console.log("Detected overworld.");
+            pbf_press_button(context, BUTTON_MINUS, 20, 100); // open pokedex
+            continue;
+        case 1:
+            console.log("Detected pokedex.");
+            return;
+        default:
+            throw OperationFailedException(
+                ErrorReport::SEND_ERROR_REPORT, console,
+                "open_pokedex_from_overworld(): No recognized state after 30 seconds.",
+                true
+            );
+        }
+    }
+}
+
+
+void open_recently_battled_from_pokedex(const ProgramInfo& info, ConsoleHandle& console, BotBaseContext& context){
+    WallClock start = current_time();
+    while (true){

```

Comment:
I think this while is executed once only (then it's stuck in the other while until it return)

Reply:
Moved time check inside the same while loop